src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -188,6 +188,19 @@ internal int CurrentPasteOffset
         /// </summary>
         private bool workspaceLoaded;
 
+        /// <summary>
+        /// This event is to trigger the DependencyRegen on the WorkspaceDependencyview extension.
+        /// </summary>
+        public static event Action TriggerDependencyRegen;
```

Comment:
I think the event should not be way to signal just that we want to trigger some other method, but more about the state of the workspace - 

like

`DummyNodesResolved`

why is this static- is there a good reason?

Reply:
Renamed the event name based on the state and made it non-static.

Reply:
Actually, this has to be static. Because when this event is attached to the current workspace in here (https://github.com/DynamoDS/Dynamo/pull/10011/files#diff-14bfe53b9085e8f1b47772149559d0a7R135), the workspace object is different. So had to keep it static for now.

Reply:
@reddyashish couldn't you just attach this handler when the workspace changes?